datafusion/core/tests/fuzz_cases/pruning.rs

```diff
@@ -110,13 +110,27 @@ async fn test_utf8_not_like_prefix() {
         .await;
 }
 
+#[tokio::test]
+async fn test_utf8_not_like_ecsape() {
+    Utf8Test::new(|value| col("a").not_like(lit(format!("\\%{}%", value))))
+        .run()
+        .await;
+}
```

Comment:
Love these fuzz tests! These 3 lines of code give me great confidence that this PR does the right thing 😄